src/material/dialog/dialog-container.ts

```diff
@@ -117,6 +126,59 @@ export class MatDialogContainer extends CdkDialogContainer<MatDialogConfig> impl
     );
   }
 
+  /** Get userAgent to check for useragent operating system */
+  private _getUserPlatform = (): string => {
+    const window = this._getWindow();
+    let userAgent = window.navigator.userAgent.toLowerCase(),
+      macosPlatforms = /(macintosh|macintel|macppc|mac68k|macos)/i,
+      windowsPlatforms = /(win32|win64|windows|wince)/i,
+      iosPlatforms = /(iphone|ipad|ipod)/i,
+      os = '';
+    if (macosPlatforms.test(userAgent)) {
+      os = 'macos';
+    } else if (iosPlatforms.test(userAgent)) {
+      os = 'ios';
+    } else if (windowsPlatforms.test(userAgent)) {
+      os = 'windows';
+    } else if (/android/.test(userAgent)) {
+      os = 'android';
+    } else if (!os && /linux/.test(userAgent)) {
+      os = 'linux';
+    }
+    return os;
+  };
+
+  /** Get Dialog name from aria attributes */
+  private _getDialogName = (): string => {
```

Comment:
This method seems pretty crazy and I would hope we can find a better way to accomplish this.

Also just in general, its bad practice to have a method called `getXYZ` that you rely on for its side effects rather than its return value.

Reply:
Thanks for the input, Miles! Revising now.

Reply:
Made adjustments based on your suggestions @mmalerba. Let me know if the updates are sufficient.